src/core_test/mpas_test_openacc.F

```diff
@@ -0,0 +1,362 @@
+                                 edge_arr_gpu)
+
+        use mpas_kind_types, only : RKIND
+
+        implicit none
+
+        ! arguments
+        integer, intent(in) :: nCells, nEdges, nCellsSolve, nEdgesSolve
+        real (kind=RKIND), dimension(:), intent(in) :: areaCell
+        integer, dimension(:), intent(in) :: indexToCellID
+        integer, dimension(:), intent(in) :: nEdgesOnCell
+        integer, dimension(:,:), intent(in) :: cellsOnEdge
+        real (kind=RKIND), dimension(:), intent(inout) :: edge_arr_gpu
+
+        ! locals
+        integer :: iCell, iEdge, cell1, cell2
+        real (kind=RKIND), dimension(nCells) :: invArea
+        real (kind=RKIND), dimension(nCells) :: help_arr
+
+        !$acc enter data copyin(nCells,nEdges, &
+        !$acc                   areaCell(:), indexToCellID(:), &
+        !$acc                   nEdgesOnCell(:),cellsOnEdge(:,:))
+
+        !$acc enter data create(edge_arr_gpu(:),iCell,iEdge,cell1,cell2, &
+        !$acc                   invArea(:),help_arr(:))
+
+        ! Compute any helpers and initialize arrs
+        !$acc parallel default(present) async
+        !$acc loop gang worker vector
+        do iCell=1,nCells
+           invArea(iCell) = 1.0_RKIND / areaCell(iCell)
+           help_arr(iCell) = 0.0_RKIND
+        end do
+
+        !$acc loop gang worker vector
+        do iEdge=1,nEdges
+           edge_arr_gpu(iEdge) = 0.0_RKIND
+        end do
+        !$acc end parallel
+
+        ! Compute helper values (for all owned cells)
+        !$acc parallel default(present) wait
+        !$acc loop gang worker vector
+        do iCell=1,nCellsSolve
+           help_arr(iCell) = (nEdgesOnCell(iCell)+indexToCellID(iCell)) * invArea(iCell)
+        end do
+        !$acc end parallel
+
+        ! Compute final value (for all owned edges)
+        !$acc parallel default(present) wait
+        !$acc loop gang worker vector private(cell1, cell2)
+        do iEdge=1,nEdgesSolve
+           cell1 = cellsOnEdge(1,iEdge)
+           cell2 = cellsOnEdge(2,iEdge)
+
+           edge_arr_gpu(iEdge) = 0.5_RKIND * (help_arr(cell1) + help_arr(cell2))
+        end do
+        !$acc end parallel
+
+        !$acc exit data delete(nCells,nEdges, &
+        !$acc                  areaCell(:), indexToCellID(:), &
+        !$acc                  nEdgesOnCell(:),cellsOnEdge(:,:), &
+        !$acc                  iCell,iEdge,cell1,cell2,invArea(:),help_arr(:))
+
+        !$acc exit data copyout(edge_arr_gpu(:))
+
+    end subroutine rep_arrs_work_gpu
+
+
+    !-----------------------------------------------------------------------
+    !  routine diff_arrs
+    !
+    !> \brief Calculate the absolute difference of two edge-based arrays
+    !> \author G. Dylan Dickerson
+    !> \date   29 May 2024
+    !> \details
+    !>  Given a cpu and a gpu version of the same edge-based array, this
+    !>  this routine uses OpenACC loops to calculate the absolute difference
```

Comment:
I agree, CPU seems like the best place for this (reduce complexity where possible). This probably came from a wrongly recorded note to self.

Reply:
For the diff between the 1-d arrays on the CPU, which would be better in Fortran?

```fortran
diff = sum(abs(arr_cpu(1:nVals) - arr_gpu(1:nVals)))
```

or

```fortran
diff = sum(abs(arr_cpu(:) - arr_gpu(:)))
```

The former seems safer, but would require one more argument or a call to `size`.

Reply:
Specifying the range of indices is probably safer. Note that there is a "garbage edge" at index `nEdges+1` for Registry-defined fields dimensioned by `nEdge`. I think we initialize the garbage edge, but it's probably not a guarantee. So, the difference between `arr_cpu(nEdges+1)` and `arr_gpu(nEdges+1)` could be nonzero if the garbage edge isn't initialized.

Reply:
Since the `diff_arrs_1d` subroutine is essentially one line of code, I think there's a good argument for just eliminating that subroutine altogether and calculating the difference inline where we would have otherwise called `diff_arrs_1d`.